rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp

```diff
@@ -158,9 +154,13 @@ std::shared_ptr<rosbag2_storage::SerializedBagMessage> SequentialReader::read_ne
 {
   if (storage_) {
     // performs rollover if necessary
-    has_next();
-    auto message = storage_->read_next();
-    return converter_ ? converter_->convert(message) : message;
+    if (has_next()) {
+      auto message = storage_->read_next();
+      // update seek time
+      seek_time_ = message->time_stamp;
```

Comment:
We should not use time_stamp from read_next message as seek_time. 
Seek time should be used only directly from `void SequentialReader::seek(const rcutils_time_point_value_t & timestamp)` API call.

Reply:
this is to track time as we proceed through the read. If we set/reset filter, we want to do so "in place" without resetting the seek time back to a previous place in time.

Reply:
@lihui815 Thanks for the explanation, I've figured it out.
 But still have opinion that we shouldn't use seek time as a filter.